src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -187,16 +189,22 @@ public BitbucketCommit call() throws Exception {
 
     private final BitbucketServerWebhookImplementation webhookImplementation;
 
+    @Deprecated
     public BitbucketServerAPIClient(@NonNull String baseURL, @NonNull String owner, @CheckForNull String repositoryName,
-        @CheckForNull StandardUsernamePasswordCredentials creds, boolean userCentric) {
-        this(baseURL, owner, repositoryName, creds, userCentric, BitbucketServerEndpoint.findWebhookImplementation(baseURL));
+                                    @CheckForNull StandardUsernamePasswordCredentials credentials, boolean userCentric) {
+        this(baseURL, owner, repositoryName, credentials != null ? new BitbucketUsernamePasswordAuthenticator(credentials) : null,
+            userCentric, BitbucketServerEndpoint.findWebhookImplementation(baseURL));
     }
 
     public BitbucketServerAPIClient(@NonNull String baseURL, @NonNull String owner, @CheckForNull String repositoryName,
```

Comment:
Missing `Deprecated`

Reply:
That was a deliberate decision. With this not being deprecated, I was able to remove all changes to `BitbucketApiFactory`. I'd rather like to make the other public constructor package private VisibleForTesting, since it is only used by the other ctors and the tests. WDYT?